src/toil/common.py

```diff
@@ -1014,26 +1017,55 @@ def _setupAutoDeployment(self, userScript=None):
             logger.debug('Injecting user script %s into batch system.', userScriptResource)
             self._batchSystem.setUserScript(userScriptResource)
 
-    def importFile(self, srcUrl, sharedFileName=None, symlink=False):
+    def importFile(self,
+                   srcUrl: str,
+                   sharedFileName: Optional[str] = None,
+                   symlink: bool = False) -> Optional[Union[FileID, str]]:
         """
         Imports the file at the given URL into job store.
 
         See :func:`toil.jobStores.abstractJobStore.AbstractJobStore.importFile` for a
         full description
         """
         self._assertContextManagerUsed()
+        srcUrl = self.sanity_check_uri_and_format(srcUrl, check_existence=True)
         return self._jobStore.importFile(srcUrl, sharedFileName=sharedFileName, symlink=symlink)
 
-    def exportFile(self, jobStoreFileID, dstUrl):
+    def exportFile(self, jobStoreFileID: Union[FileID, str], dstUrl: str) -> None:
         """
         Exports file to destination pointed at by the destination URL.
 
         See :func:`toil.jobStores.abstractJobStore.AbstractJobStore.exportFile` for a
         full description
         """
         self._assertContextManagerUsed()
+        dstUrl = self.sanity_check_uri_and_format(dstUrl)
         self._jobStore.exportFile(jobStoreFileID, dstUrl)
 
+    @staticmethod
+    def sanity_check_uri_and_format(uri: str, check_existence: bool = False) -> str:
+        """
+        Transforms local paths into absolute local paths and ensures the file:// schema is present.
+
+        Ignores URIs with other schema (examples: s3:/, gs:/, ftp://, etc.).
+        """
+        if urlparse(uri).scheme == 'file':
+            uri = urlparse(uri).path  # this should strip off the local file schema; it will be added back
+
+        # account for the schema-less case, which should be coerced to a local absolute path
+        if urlparse(uri).scheme == '':
+            abs_path = os.path.abspath(uri)
+            if not os.path.exists(abs_path) and check_existence:
+                raise FileNotFoundError(
+                    f'Import failed for: "{uri}"\n'
+                    f'No schema (http://, s3://, file://, etc.) for path was included.\n'
```

Comment:
Isn't it "scheme" and not "schema"?

Reply:
Also, didn't we just drop the `file:` scheme that may have been included?

Reply:
Hmmm... I've been using them interchangeably.  I'll switch over to scheme to be consistent with urlparse.  There's an interesting suggestion here that schema's modern usage is largely Kant's fault: https://english.stackexchange.com/questions/40702/difference-between-scheme-and-schema

And yes, the `file:` scheme was dropped so that the same scheme-less conditional would modify it to be an absolute path and then add `file://` back.